src/Core/Domain/Carrier/ValueObject/CarrierRangesCollection.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Carrier\ValueObject;
+
+use IteratorAggregate;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\Exception\CarrierConstraintException;
+
+class CarrierRangesCollection implements CarrierRangesCollectionInterface
+{
+    public function __construct(
+        private int $calculatingMethod,
+        private array $zones,
+    ) {
+    }
+
+    public function getCalculatingMethod(): int
+    {
+        return $this->calculatingMethod;
+    }
+
+    public function getZones(): array
+    {
+        return $this->zones;
+    }
+
+    /**
+     * @param int $calculatingMethod
+     * @param IteratorAggregate|array $carrierRanges
+     *
+     * @return CarrierRangesCollection
+     *
+     * @throws CarrierConstraintException
+     */
+    public static function createFromData(int $calculatingMethod, IteratorAggregate|array $carrierRanges): CarrierRangesCollection
```

Comment:
Why not simply handle this in the `__constructor`?

This way we can do `$ranges = new CarrierRangesCollection($carrier->getShippingMethod(), $node);` which will be much more adapted for the API integration later